src/fprime_extras/lint/rules/rule_valid_port_types.py

```diff
@@ -0,0 +1,156 @@
+import logging
+
+from ...core.hooks import HookRegistry as hooks
+from .base import AbstractRule
+from .base import LintSeverity
+from .base import Notification
+from .registry import RuleListRegistry
+from ...core.util import find_fprime
+
+from pathlib import Path
+import os, sys
+import lxml.etree as etree
+from lxml.etree import XMLSyntaxError
+
+log = logging.getLogger(__name__)
+port_slug = 'invalid-port-type'
+root_slug = 'invalid-fprime-structure'
+import_types_slug = 'invalid-port-type-import'
+# TODO: import following list from model instead of defining it here.
+port_kinds = ['output', 'sync_input', 'async_input', 'guarded_input']
```

Comment:
This line is not used, correct?  Perhaps delete it and the ``TODO`` statement.  Copy Pasta.

Reply:
I deleted this list